Library/Homebrew/dev-cmd/pull.rb

```diff
@@ -237,40 +243,72 @@ def pull
         safe_system "git", "commit", "--amend", "--signoff", "--allow-empty", "-q", "-m", message
       end
 
+      if testing_job
+        bottle_branch = "testing-bottle-#{testing_job}"
+      else
+        bottle_branch = "pull-bottle-#{issue}"
+      end
+
       # Bottles: Pull bottle block commit and publish bottle files on Bintray
       if fetch_bottles
         bottle_commit_url = if testing_job
-          bottle_branch = "testing-bottle-#{testing_job}"
           url
         else
-          bottle_branch = "pull-bottle-#{issue}"
           bot_username = GitHub.test_bot_user(user, args.test_bot_user)
           "https://github.com/#{bot_username}/homebrew-#{tap.repo}/compare/#{user}:master...pr-#{issue}"
         end
 
         curl "--silent", "--fail", "--output", "/dev/null", "--head", bottle_commit_url
 
-        safe_system "git", "checkout", "--quiet", "-B", bottle_branch, orig_revision
-        pull_patch bottle_commit_url, "bottle commit"
-        safe_system "git", "rebase", "--quiet", branch
-        safe_system "git", "checkout", "--quiet", branch
-        safe_system "git", "merge", "--quiet", "--ff-only", "--no-edit", bottle_branch
-        safe_system "git", "branch", "--quiet", "-D", bottle_branch
+        if merge_commit
+          PatchPuller.new(bottle_commit_url, args, "merge commit").pull_merge_commit(issue)
+        else
+          safe_system "git", "checkout", "--quiet", "-B", bottle_branch, orig_revision
+          PatchPuller.new(bottle_commit_url, args, "bottle commit").pull_patch
+          safe_system "git", "rebase", "--quiet", branch
+          safe_system "git", "checkout", "--quiet", branch
+          safe_system "git", "merge", "--quiet", "--ff-only", "--no-edit", bottle_branch
+          safe_system "git", "branch", "--quiet", "-D", bottle_branch
+        end
 
         # Publish bottles on Bintray
         unless args.no_publish?
           published = publish_changed_formula_bottles(tap, changed_formulae_names)
           bintray_published_formulae.concat(published)
         end
+
+        # Verify bintray publishing after all patches have been applied
+        bintray_published_formulae.uniq!
+        verify_bintray_published(bintray_published_formulae)
+
+        ohai "Published bottles for:"
+        puts bintray_published_formulae.join " "
+      end
+
+      if merge_commit && !fetch_bottles
+        ohai "Fast-forwarding to the merge commit"
+        safe_system "git", "checkout", "--quiet", "-B", bottle_branch, orig_revision
+        fetch_issue(url)
+        system "git", "merge", "--quiet", "--ff-only", "--no-edit", "FETCH_HEAD"
```

Comment:
Why is this one not `safe_system`?

Reply:
That's a copy/paste typo. I'll change that.